### PR TITLE
Modify run_e2etest script to check for job_types and include_dirs

### PR DESCRIPTION
**Pull request by richardsliu**
Related to #150 run E2E tests selectively based on modified source files.
Related to #168 project velocity negatively impacted by test flakiness.

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/180)

### CODE REVIEW

**Comment by jlewi** [comment]
This looks great.

Can you update README.md to provide instructions on how to scope tests? In particular can you include a snippet of what the update YAML looks like?

**Comment by richardsliu** [comment]
Done.

File: README.md [hunk 1]
```diff
@@ -170,6 +170,18 @@ configured for the repository (see these [instructions](#prow-setup) for info on
...
    * Add a `prow_config.yaml` file that will be passed into run_e2e_workflow to determine which ksonnet app to use for testing. An example can be seen [here](https://github.com/kubeflow/kubeflow/blob/master/prow_config.yaml).
+     * Workflows can optionally be scoped by job type (presubmit/postsubmit) or modified directories. For example:
+
+       ```
+       workflows:
+        - app_dir: kubeflow/testing/workflows
+          component: workflows
+          name: unittests
+          job_types: presubmit
+          include_dirs: foo/*,bar/*
```
Comment (jlewi) [inline README.md]: job_types and include_dirs should be arrays of strings e.g.

include_dirs:
  - foo/*
  - bar/*

**Comment by richardsliu** [comment]
Done. I also removed testMainPresubmit since it no longer applies.

**Comment by jlewi** [comment]
This is fantastic.

/lgtm
/approve

**Pull request merged**